File: PST/FUEL/src/FUEL/ctrlUserLocation.vb
Imports DevComponents.DotNetBar
Imports DevComponents.DotNetBar.Controls
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace FUEL
    <DesignerGenerated> _
    Public Class ctrlUserLocation
        Inherits UserControl
        ' Methods
        <DebuggerNonUserCode> _
        Public Sub New()
            AddHandler MyBase.Load, New EventHandler(AddressOf Me.ctrlUserLocation_Load)
            ctrlUserLocation.__ENCAddToList(Me)
            Me.InitializeComponent
        End Sub

        <DebuggerNonUserCode> _
        Private Shared Sub __ENCAddToList(ByVal value As Object)
            SyncLock ctrlUserLocation.__ENCList
                If (ctrlUserLocation.__ENCList.Count = ctrlUserLocation.__ENCList.Capacity) Then
                    Dim index As Integer = 0
                    Dim num3 As Integer = (ctrlUserLocation.__ENCList.Count - 1)
                    Dim num2 As Integer = 0
                    Do While True
                        Dim num4 As Integer = num3
                        If (num2 > num4) Then
                            ctrlUserLocation.__ENCList.RemoveRange(index, (ctrlUserLocation.__ENCList.Count - index))
                            ctrlUserLocation.__ENCList.Capacity = ctrlUserLocation.__ENCList.Count
                            Exit Do
                        End If
                        Dim reference As WeakReference = ctrlUserLocation.__ENCList(num2)
                        If reference.IsAlive Then
                            If (num2 <> index) Then
                                ctrlUserLocation.__ENCList(index) = ctrlUserLocation.__ENCList(num2)
                            End If
                            index += 1
                        End If
                        num2 += 1
                    Loop
                End If
                ctrlUserLocation.__ENCList.Add(New WeakReference(value))
            End SyncLock
        End Sub

        Private Sub ctrlUserLocation_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim enumerator As IEnumerator
            Try 
                enumerator = Enum.GetValues(GetType(TestSites)).GetEnumerator
                Do While True
                    If Not enumerator.MoveNext Then
                        Exit Do
                    End If
                    Dim current As Object = enumerator.Current
                    Me.cboSiteList.Items.Add(current.ToString)
                Loop
            Finally
                If Not Object.ReferenceEquals(TryCast(enumerator,IDisposable), Nothing) Then
                    TryCast(enumerator,IDisposable).Dispose
                End If
            End Try
        End Sub

        <DebuggerNonUserCode> _
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            Try 
                If (If((Not disposing OrElse (Me.components Is Nothing)), 0, 1) <> 0) Then
                    Me.components.Dispose
                End If
            Finally
                MyBase.Dispose(disposing)
            End Try
        End Sub

        <DebuggerStepThrough> _
        Private Sub InitializeComponent()
            Me.cboSiteList = New ComboBoxEx
            Me.LabelX1 = New LabelX
            Me.LabelX2 = New LabelX
            Me.SuspendLayout
            Me.cboSiteList.Anchor = (AnchorStyles.Right Or (AnchorStyles.Left Or AnchorStyles.Top))
            Me.cboSiteList.DisplayMember = "Text"
            Me.cboSiteList.DrawMode = DrawMode.OwnerDrawFixed
            Me.cboSiteList.FormattingEnabled = True
            Me.cboSiteList.ItemHeight = 14
            Dim point2 As New Point(&H10, &H3E)
            Me.cboSiteList.Location = point2
            Me.cboSiteList.Name = "cboSiteList"
            Dim size2 As New Size(&HF5, 20)
            Me.cboSiteList.Size = size2
            Me.cboSiteList.Style = eDotNetBarStyle.StyleManagerControlled
            Me.cboSiteList.TabIndex = 0
            Me.LabelX1.Anchor = (AnchorStyles.Right Or (AnchorStyles.Left Or AnchorStyles.Top))
            Me.LabelX1.BackgroundStyle.CornerType = eCornerType.Square
            point2 = New Point(&H10, &H26)
            Me.LabelX1.Location = point2
            Me.LabelX1.Name = "LabelX1"
            size2 = New Size(&HF5, &H17)
            Me.LabelX1.Size = size2
            Me.LabelX1.TabIndex = 1
            Me.LabelX1.Text = "Please Select Your Location from the List Below"
            Me.LabelX2.Anchor = (AnchorStyles.Right Or (AnchorStyles.Left Or AnchorStyles.Top))
            Me.LabelX2.BackgroundStyle.CornerType = eCornerType.Square
            point2 = New Point(&H10, 3)
            Me.LabelX2.Location = point2
            Me.LabelX2.Name = "LabelX2"
            size2 = New Size(&HF5, &H17)
            Me.LabelX2.Size = size2
            Me.LabelX2.TabIndex = 2
            Me.LabelX2.Text = "Before proceeding, I must know your location."
            Dim ef2 As New SizeF(6!, 13!)
            Me.AutoScaleDimensions = ef2
            Me.AutoScaleMode = AutoScaleMode.Font
            Me.Controls.Add(Me.LabelX2)
            Me.Controls.Add(Me.LabelX1)
            Me.Controls.Add(Me.cboSiteList)
            Me.Name = "ctrlUserLocation"
            size2 = New Size(280, &H63)
            Me.Size = size2
            Me.ResumeLayout(False)
        End Sub


        ' Properties
        Friend Overridable Property cboSiteList As ComboBoxEx
            <DebuggerNonUserCode> _
            Get
                Return Me._cboSiteList
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As ComboBoxEx)
                Me._cboSiteList = WithEventsValue
            End Set
        End Property

        Friend Overridable Property LabelX1 As LabelX
            <DebuggerNonUserCode> _
            Get
                Return Me._LabelX1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As LabelX)
                Me._LabelX1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property LabelX2 As LabelX
            <DebuggerNonUserCode> _
            Get
                Return Me._LabelX2
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As LabelX)
                Me._LabelX2 = WithEventsValue
            End Set
        End Property


        ' Fields
        Private Shared __ENCList As List(Of WeakReference) = New List(Of WeakReference)
        Private components As IContainer
        <AccessedThroughProperty("cboSiteList")> _
        Private _cboSiteList As ComboBoxEx
        <AccessedThroughProperty("LabelX1")> _
        Private _LabelX1 As LabelX
        <AccessedThroughProperty("LabelX2")> _
        Private _LabelX2 As LabelX
    End Class
End Namespace

File: PST/FUEL Source Code/FUEL/Backup/ctrlUserLocation.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.ctrlUserLocation
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using DevComponents.DotNetBar;
using DevComponents.DotNetBar.Controls;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Drawing;
using System.Runtime.CompilerServices;
using System.Windows.Forms;

namespace FUEL
{
  [DesignerGenerated]
  public class ctrlUserLocation : UserControl
  {
    private static List<WeakReference> __ENCList = new List<WeakReference>();
    private IContainer components;
    [AccessedThroughProperty("cboSiteList")]
    private ComboBoxEx _cboSiteList;
    [AccessedThroughProperty("LabelX1")]
    private LabelX _LabelX1;
    [AccessedThroughProperty("LabelX2")]
    private LabelX _LabelX2;

    [DebuggerNonUserCode]
    static ctrlUserLocation()
    {
    }

    [DebuggerNonUserCode]
    public ctrlUserLocation()
    {
      this.Load += new EventHandler(this.ctrlUserLocation_Load);
      ctrlUserLocation.__ENCAddToList((object) this);
      this.InitializeComponent();
    }

    [DebuggerNonUserCode]
    private static void __ENCAddToList(object value)
    {
      lock (ctrlUserLocation.__ENCList)
      {
        if (ctrlUserLocation.__ENCList.Count == ctrlUserLocation.__ENCList.Capacity)
        {
          int index1 = 0;
          int num = checked (ctrlUserLocation.__ENCList.Count - 1);
          int index2 = 0;
          while (index2 <= num)
          {
            if (ctrlUserLocation.__ENCList[index2].IsAlive)
            {
              if (index2 != index1)
                ctrlUserLocation.__ENCList[index1] = ctrlUserLocation.__ENCList[index2];
              checked { ++index1; }
            }
            checked { ++index2; }
          }
          ctrlUserLocation.__ENCList.RemoveRange(index1, checked (ctrlUserLocation.__ENCList.Count - index1));
          ctrlUserLocation.__ENCList.Capacity = ctrlUserLocation.__ENCList.Count;
        }
        ctrlUserLocation.__ENCList.Add(new WeakReference(RuntimeHelpers.GetObjectValue(value)));
      }
    }

    [DebuggerNonUserCode]
    protected override void Dispose(bool disposing)
    {
      try
      {
        if (!disposing || this.components == null)
          return;
        this.components.Dispose();
      }
      finally
      {
        base.Dispose(disposing);
      }
    }

    [DebuggerStepThrough]
    private void InitializeComponent()
    {
      this.cboSiteList = new ComboBoxEx();
      this.LabelX1 = new LabelX();
      this.LabelX2 = new LabelX();
      this.SuspendLayout();
      this.cboSiteList.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;
      this.cboSiteList.DisplayMember = "Text";
      this.cboSiteList.DrawMode = DrawMode.OwnerDrawFixed;
      this.cboSiteList.FormattingEnabled = true;
      this.cboSiteList.ItemHeight = 14;
      ComboBoxEx cboSiteList1 = this.cboSiteList;
      Point point1 = new Point(16, 62);
      Point point2 = point1;
      cboSiteList1.Location = point2;
      this.cboSiteList.Name = "cboSiteList";
      ComboBoxEx cboSiteList2 = this.cboSiteList;
      Size size1 = new Size(245, 20);
      Size size2 = size1;
      cboSiteList2.Size = size2;
      this.cboSiteList.Style = eDotNetBarStyle.StyleManagerControlled;
      this.cboSiteList.TabIndex = 0;
      this.LabelX1.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;
      this.LabelX1.BackgroundStyle.CornerType = eCornerType.Square;
      LabelX labelX1_1 = this.LabelX1;
      point1 = new Point(16, 38);
      Point point3 = point1;
      labelX1_1.Location = point3;
      this.LabelX1.Name = "LabelX1";
      LabelX labelX1_2 = this.LabelX1;
      size1 = new Size(245, 23);
      Size size3 = size1;
      labelX1_2.Size = size3;
      this.LabelX1.TabIndex = 1;
      this.LabelX1.Text = "Please Select Your Location from the List Below";
      this.LabelX2.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;
      this.LabelX2.BackgroundStyle.CornerType = eCornerType.Square;
      LabelX labelX2_1 = this.LabelX2;
      point1 = new Point(16, 3);
      Point point4 = point1;
      labelX2_1.Location = point4;
      this.LabelX2.Name = "LabelX2";
      LabelX labelX2_2 = this.LabelX2;
      size1 = new Size(245, 23);
      Size size4 = size1;
      labelX2_2.Size = size4;
      this.LabelX2.TabIndex = 2;
      this.LabelX2.Text = "Before proceeding, I must know your location.";
      this.AutoScaleDimensions = new SizeF(6f, 13f);
      this.AutoScaleMode = AutoScaleMode.Font;
      this.Controls.Add((Control) this.LabelX2);
      this.Controls.Add((Control) this.LabelX1);
      this.Controls.Add((Control) this.cboSiteList);
      this.Name = nameof (ctrlUserLocation);
      size1 = new Size(280, 99);
      this.Size = size1;
      this.ResumeLayout(false);
    }

    internal virtual ComboBoxEx cboSiteList
    {
      [DebuggerNonUserCode] get => this._cboSiteList;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._cboSiteList = value;
    }

    internal virtual LabelX LabelX1
    {
      [DebuggerNonUserCode] get => this._LabelX1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._LabelX1 = value;
    }

    internal virtual LabelX LabelX2
    {
      [DebuggerNonUserCode] get => this._LabelX2;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._LabelX2 = value;
    }

    private void ctrlUserLocation_Load(object sender, EventArgs e)
    {
      try
      {
        foreach (object obj in Enum.GetValues(typeof (PST.TestSites)))
          this.cboSiteList.Items.Add((object) RuntimeHelpers.GetObjectValue(obj).ToString());
      }
      finally
      {
        IEnumerator enumerator;
        if (enumerator is IDisposable)
          (enumerator as IDisposable).Dispose();
      }
    }
  }
}
